src/FSharpVSPowerTools.Logic/GoToDefinitionFilter.fs

```diff
@@ -421,10 +432,21 @@ type GoToDefinitionFilter(textDocument: ITextDocument,
                 match symbolResult with
                 | Some (_, _, _, _, FSharpFindDeclResult.DeclFound _) 
                 | None ->
-                    // Run the operation on UI thread since continueCommandChain may access UI components.
-                    do! Async.SwitchToContext uiContext
-                    // Declaration location might exist so let Visual F# Tools handle it. 
-                    return continueCommandChain()
+                    // no FSharpSymbol found, here we look at #load directive
```

Comment:
I fixed all other remarks.

I'll give a look at this tomorrow. I didn't change ```getDocumentState``` (beside extracting some code to get filepath, project and document which is also used with the changes in ```gotoDefinition```) and would certainly see changing it's return type be relevant but felt like too much overhead for my change when I was fixing ```goToDefinition```.

I'll spend time tomorrow and will implement the refactoring you propose, thanks for giving me a clue about expected return type!